src/network_manager.rs

```diff
@@ -88,13 +93,24 @@ where
             })
             .and_then(|written| {
                 if written != packet.len() {
-                    Err(Error::WriteFail)
+                    self.unfinished_packet = Some(Vec::from_slice(&packet[written..]).unwrap());
```

Comment:
I don't think the user actually cares if a partial write occurs. Ultimately, we'll come back around and finish the write as soon as we have space in the network stack. Just return `Ok(())` here, since we'll finish the write later.

Reply:
```rust
self.unfinished_packet.replace(Vec::from_slice(&packet[written..]).unwrap());
```